src/ifcgeom/mapping/mapping.cpp

```diff
@@ -565,11 +565,13 @@ taxonomy::ptr mapping::map_impl(const IfcSchema::IfcStyledItem* inst) {
         return surface_style;
     }
 
-    static taxonomy::colour white = taxonomy::colour(1., 1., 1.);
-    double rgb[3];
-    if (process_colour(shading->SurfaceColour(), rgb)) {
-        surface_style->diffuse.components() << rgb[0], rgb[1], rgb[2];
-    }
+	surface_style->use_surface_color = settings_.get<settings::SurfaceColour>().get();
+
+	static taxonomy::colour white = taxonomy::colour(1., 1., 1.);
+	double rgb[3];
+	if (process_colour(shading->SurfaceColour(), rgb)) {
+		surface_style->surface.components() << rgb[0], rgb[1], rgb[2];
+	}
 
     if (auto rendering_style = shading->as<IfcSchema::IfcSurfaceStyleRendering>()) {
         if (rendering_style->DiffuseColour() && process_colour(rendering_style->DiffuseColour(), rgb)) {
```

Comment:
I cannot make a comment in Gh about the line below :(

`576 | const taxonomy::colour& old_diffuse = surface_style->diffuse ? surface_style->diffuse : white;`

Shouldn't that also be changed to `surface_style->surface`? To retain the old behaviour of the diffuse value?

Reply:
You say in this case, add a check so that if there is no value in the surface, put diffuse, right?
I'll add a check with `if (!surface_style->diffuse)` or something.

Reply:
But let me comment. In get_color I have checked so that when there is no surface, it gets diffuse. 
Or when there was no diffuse, use the surface (I don't know if this is a possibility)

```
const colour& get_color() const {
	if ((use_surface_color && **surface**) || !diffuse) {
		return surface;
	}
	return diffuse;
}
```